chain/chain/src/store.rs

```diff
@@ -3221,8 +3223,8 @@ mod tests {
             let mut store_update = chain.mut_store().store_update();
             store_update.save_block(block.clone());
             store_update.inc_block_refcount(block.header().prev_hash()).unwrap();
-            store_update.save_head(&Tip::from_header(block.header())).unwrap();
             store_update.save_block_header(block.header().clone()).unwrap();
```

Comment:
so the ordering here is important ? was it a bug before ?

Reply:
should we add something (in separate PR) to enforce the proper ordering ?

Reply:
this is a piece of testing code